File: php/6-zigzag-conversion/zigzag-conversion.php
<?php

function convert($s, $numRows) {
    if ($numRows < 0 || strlen($s) <= $numRows) {
        return "";
    }

    $rows = array_fill(0, $numRows, "");

    $currentRow = 0;
    $goingDown = true;

    for ($i = 0; $i < strlen($s); $i++) {

        $char = $s[$i];

        $rows[$currentRow] .= $char;

        if ($currentRow == $numRows) {
            $goingDown = false;
        }

        if ($goingDown) {
            $currentRow++;
        } else {
            $currentRow--;
        }

        if ($currentRow < 0) {
            $currentRow = 0;
            $goingDown = true;
        }
    }

    $result = "";

    for ($i = count($rows) - 1; $i >= 0; $i--) {
        $result .= $rows[$i];
    }

    return trim($result);
}

?>

File: kotlin/6-zigzag-conversion/zigzag-conversion.kt
class Solution {
    fun convert(s: String, numRows: Int): String {

if (numRows == 1 || s.length <= numRows) {
    return s
}
            return ""
        }

        val rows = Array(numRows) { StringBuilder() }

        var currentRow = 0
        var goingDown = true

        for (i in 0 until s.length) {

            val char = s[i]

            rows[currentRow].append(char)

// Change direction at the top or bottom row
if (currentRow == 0) {
    goingDown = true
} else if (currentRow == numRows - 1) {
    goingDown = false
}
                goingDown = false
            }

            if (goingDown) {
                currentRow++
            } else {
                currentRow--
            }

            if (currentRow < 0) {
                currentRow = 0
                goingDown = true
            }
        }

        val result = StringBuilder()

        for (i in rows.size - 1 downTo 0) {
            result.append(rows[i])
        }

        return result.toString().trim()
    }
}